### ISSUE TITLE
Flutter Hot Reload will not work

### ISSUE THREAD

**Issue by FaisalAbid** [flutter/flutter#17755]
## Steps to Reproduce

Trying to hot reload. App runs properly on first build, but running hot reload after causes an error.

## Logs

```

Initializing hot reload...
../../third_party/dart/runtime/vm/object.h: 8818: error: Handle check failed: saw Instance of 'MaterialApp' expected Closure
Dumping native stack trace for thread 5a03
[0x000000010b8154c4] dart::Profiler::DumpStackTrace(void*)
[0x000000010b8154c4] dart::Profiler::DumpStackTrace(void*)
[0x000000010ba889a2] dart::Assert::Fail(char const*, ...)
[0x000000010b86d31b] dart::DRT_InvokeClosureNoSuchMethod(dart::NativeArguments)
[0x000000012b24085b] [Stub] CallToRuntime
[0x000000012b242dc4] [Stub] CallClosureNoSuchMethod
[0x000000013465e8de] VenChatAppState.build
[0x000000013363ca14] StatefulElement.build
[0x000000013363bd5e] ComponentElement.performRebuild
[0x000000013363aec1] Element.rebuild
[0x00000001336164f0] BuildOwner.buildScope
[0x0000000135873230] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame
[0x0000000135872e28] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback
[0x0000000135872c3f] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback
[0x000000012d40dd15] _Closure.call
[0x0000000135872476] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback
[0x00000001358714a8] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame
[0x0000000135870feb] _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure>
[0x000000012d400792] _Closure.call
[0x000000013586c7ed] Timer._createTimer.<anonymous closure>
[0x000000012d40dd15] _Closure.call
[0x000000013586b8f5] _Timer._runTimers
[0x0000000135869b72] _Timer._handleMessage
[0x000000013586967b] _Timer._handleMessage
[0x000000012d40dd15] _Closure.call
[0x000000012d40dc1c] _RawReceivePortImpl._handleMessage
[0x000000012b240ce3] [Stub] InvokeDartCode
[0x000000010b6ce2e9] dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)
[0x000000010b6d1a2e] dart::DartLibraryCalls::HandleMessage(dart::Object const&, dart::Instance const&)
[0x000000010b6fcf66] dart::IsolateMessageHandler::HandleMessage(dart::Message*)
[0x000000010b7276ce] dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)
[0x000000010b7277d0] dart::MessageHandler::HandleNextMessage()
[0x000000010ba06d2e] Dart_HandleMessage
[0x000000010b60fad1] tonic::DartMessageHandler::OnHandleMessage(tonic::DartState*)
[0x000000010b1df386] fml::MessageLoopImpl::RunExpiredTasks()
[0x000000010b1e206c] fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*)
[0x000000010df62e64] __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
[0x000000010df62a52] __CFRunLoopDoTimer
[0x000000010df6260a] __CFRunLoopDoTimers
[0x000000010df59e4c] __CFRunLoopRun
[0x000000010df5930b] CFRunLoopRunSpecific
[0x000000010b1e21bd] fml::MessageLoopDarwin::Run()
[0x000000010b1df4bd] fml::MessageLoopImpl::DoRun()
[0x000000010b1e1bc2] std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, fml::Thread::Thread(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::$_0> >(void*, void*)
[0x0000000111d75661] _pthread_body
[0x0000000111d7550d] _pthread_body
Lost connection to device.
[0x0000000111d74bf9] thread_start
-- End of DumpStackTrace

```

Unrelated 
```
Analyzing VenFlutter...                                          

   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:2:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:4:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:6:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:8:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:10:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:12:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:14:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:16:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:18:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:20:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:22:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:24:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:26:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:28:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:30:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:32:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:34:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:36:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:38:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:40:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/constants.dart:42:14
   info • Name non-constant identifiers using lowerCamelCase • lib/api/types/conversationRequest.dart:7:9
   info • The value of the field '_ui_type' isn't used • lib/api/types/conversationRequest.dart:7:9
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:4:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:5:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:6:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:7:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:8:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:9:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:10:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:11:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:12:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:13:17
   info • Name non-constant identifiers using lowerCamelCase • lib/core/config.dart:14:17
   info • 'child' is deprecated and shouldn't be used • lib/core/widgetUtils.dart:19:7
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:10:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:11:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:12:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:13:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:14:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:15:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:16:23
   info • Name non-constant identifiers using lowerCamelCase • lib/datastore/datastore.dart:17:23
   info • 'child' is deprecated and shouldn't be used • lib/pages/chat.dart:150:36
   info • The value of the local variable 'response' isn't used • lib/pages/pointsScreen.dart:99:12
   info • This class inherits from a class marked as @immutable, and therefore should be immutable (all instance fields must be final) • lib/pages/signup.dart:12:7
   info • 'child' is deprecated and shouldn't be used • lib/pages/signup.dart:49:7
   info • This class inherits from a class marked as @immutable, and therefore should be immutable (all instance fields must be final) • lib/pages/snapscreen.dart:8:7


```

```
[✓] Flutter (Channel master, v0.4.5-pre.52, on Mac OS X 10.13.4 17E202, locale en-US)
    • Flutter version 0.4.5-pre.52 at /Users/faisalabid/Development/flutter
    • Framework revision 8717a5e6f7 (2 days ago), 2018-05-19 03:19:35 +0300
    • Engine revision 1179c38a42
    • Dart version 2.0.0-dev.55.0.flutter-43635d3372

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/faisalabid/Library/Android/sdk
    • Android NDK at /Users/faisalabid/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.3.1, Build version 9E501
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 24.2.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[!] VS Code (version 1.23.1)
    • VS Code at /Users/faisalabid/Applications/Visual Studio Code.app/Contents
    • Dart Code extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code

[✓] Connected devices (1 available)
    • iPhone X • 212BB743-F79A-4DB9-80E7-DA49B3431B0D • ios • iOS 11.3 (simulator)

! Doctor found issues in 1 category.

```

**Comment by FaisalAbid** [comment]
Turning off track widget creation fixes this

**Comment by vagrantrobbie** [comment]
how did you turn it off? I'm using IntelliJ and it keeps freezing on Hot Reload. This only started happening when I moved to Channel beta, v0.3.2

**Comment by devoncarew** [comment]
I'm seeing this as well, for the iOS simulator. /cc @jacob314

**Comment by jacob314** [comment]
The problem is --track-widget-creation combined  with the ios simulator.  I'm working to fix the issue on the IOS simulator but for now you should be able to unblock by using an android simulator or device.

**Comment by vagrantrobbie** [comment]
I'm not sure if this is the same problem but my Hot Reload issues seem to have disappeared after finding that I was getting an error from the Fluro Router TransitionType.native on iOS.

The transitions worked fine in debug but when I did a --release build the route would fail with a layout error. I forced the transition type on iOS to use TransitionType.inFromRight and I seem to have no more issues.

Like I say, I'm not 100% sure this was the cause of the bug (I didn't have track widget creation on) https://github.com/theyakka/fluro/issues/51 but I am back working fine again
